src/Compilers/CSharp/Portable/Binder/BuckStopsHereBinder.cs

```diff
@@ -15,11 +15,19 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal class BuckStopsHereBinder : Binder
     {
-        internal BuckStopsHereBinder(CSharpCompilation compilation)
+        internal BuckStopsHereBinder(CSharpCompilation compilation, SyntaxTree? associatedSyntaxTree)
             : base(compilation)
         {
+            this.AssociatedSyntaxTree = associatedSyntaxTree;
         }
 
+        /// <summary>
+        /// In non-speculative scenarios, the syntax tree being bound.
+        /// In speculative scenarios, the syntax tree from the original compilation used as the speculation context.
```

Comment:
Let's document when `AssociatedSyntaxTree` is `null` (in the binder used for conversions).